src/predefinedComponents/TabbedHeader/components/Pager.tsx

```diff
@@ -0,0 +1,321 @@
+
+    const goToPageAnimationFrame = useRef<ReturnType<typeof requestAnimationFrame>>();
+    const scrollToPageOffsetTimeout = useRef<ReturnType<typeof setTimeout>>();
+
+    const isInverted = Platform.OS === 'android' ? I18nManager.isRTL : undefined;
+
+    const offscreenPageLimitValidated = useMemo(() => {
+      return offscreenPageLimit >= DEFAULT_OFFSCREEN_PAGE_LIMIT
+        ? offscreenPageLimit
+        : DEFAULT_OFFSCREEN_PAGE_LIMIT;
+    }, [offscreenPageLimit]);
+
+    useEffect(() => {
+      /**
+       * Scroll to make first rendered tab visible (if not used, sometimes when Pager is first rendered, it has blank first tab)
+       */
+      function scrollOnePxAndBack() {
+        'worklet';
+        scrollTo(horizontalScrollViewRef, 0, 1, true);
+        scrollTo(horizontalScrollViewRef, 0, 0, true);
+      }
+
+      runOnUI(scrollOnePxAndBack)();
+
+      return () => {
+        if (goToPageAnimationFrame.current) {
+          cancelAnimationFrame(goToPageAnimationFrame.current);
+        }
+        if (scrollToPageOffsetTimeout.current) {
+          clearTimeout(scrollToPageOffsetTimeout.current);
+        }
+      };
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+      if (page !== currentPage && page >= 0) {
+        goToPage(page);
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [page]);
+
+    function onContainerLayout(e: LayoutChangeEvent) {
+      const { width } = e.nativeEvent.layout;
+
+      if (!width || width <= 0 || Math.round(width) === Math.round(containerWidth)) {
+        return;
+      }
+
+      setContainerWidth(width);
+      goToPageAnimationFrame.current = requestAnimationFrame(() => {
+        goToPage(currentPage);
+      });
+    }
+
+    function scrollToPage(offset: number) {
+      'worklet';
+      scrollTo(horizontalScrollViewRef, offset, 0, true);
+    }
+
+    function scrollToTabPosition(position: number) {
+      'worklet';
+      scrollTo(scrollRef, 0, position, true);
+    }
+
+    function goToPage(pageNumber: number) {
+      const offset = pageNumber * containerWidth;
+
+      handleScrollToTabPosition(currentPage, pageNumber);
+      runOnUI(scrollToPage)(offset);
+
+      setCurrentPage(page);
+      onChangeTab?.(currentPage, pageNumber);
+    }
+
+    function handleScrollToTabPosition(prevPage: number, newPage: number) {
+      if (!data.length || scrollValue.value === 0) {
+        return;
+      }
+
+      tabsScrollPosition.current[prevPage] = scrollValue.value;
+      scrollToPageOffsetTimeout.current = setTimeout(() => {
+        const scrollTargetPosition =
+          rememberTabScrollPosition && tabsScrollPosition.current[newPage] !== -1
+            ? tabsScrollPosition.current[newPage]
+            : scrollHeight;
+
+        runOnUI(scrollToTabPosition)(scrollTargetPosition);
+      }, SCROLL_TO_PAGE_OFFSET_TIMEOUT);
```

Comment:
As discussed with @mateusz1913 we could consider using **withDelay** instead of timeouts